action.yml

```diff
@@ -1,60 +1,188 @@
+        tutor local exec cms pip install ${{ inputs.openedx_extra_pip_requirements }}
+      shell: bash
+
+    - name: Run migrations and restart services
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor local exec lms python manage.py lms migrate
+        tutor local exec cms python manage.py cms migrate
+        tutor local restart
+      shell: bash
+
+    - name: Import Demo course
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor local do importdemocourse
+      shell: bash
+
+    - name: Test Open edX imports in plugin
+      if: ${{ inputs.openedx_imports_test_file_path && inputs.openedx_imports_test_function_name }}
+      run: |
+        source .tutor_venv/bin/activate
+
+        cp $GITHUB_ACTION_PATH/run_openedx_imports_test.py $GITHUB_WORKSPACE/${{ inputs.app_name }}/
+        echo "Running test for Open edX imports in plugin"
+
+        tutor local exec --env PYTHONPATH=/openedx/${{ inputs.app_name }} \
+                        --env DJANGO_SETTINGS_MODULE=lms.envs.production \
+                        lms python /openedx/${{ inputs.app_name }}/run_openedx_imports_test.py \
+                        --test-file-path /openedx/${{ inputs.app_name }}/${{ inputs.openedx_imports_test_file_path }} \
+                        --test-function-name '${{ inputs.openedx_imports_test_function_name }}'
+      shell: bash
+
+    - name: Load initial data for the tests
+      if: ${{ inputs.fixtures_file }}
+      run: |
+        source .tutor_venv/bin/activate
+
+        echo "Copying fixtures file to the LMS container"
+        tutor local exec lms python manage.py lms loaddata /openedx/${{ inputs.app_name }}/${{ inputs.fixtures_file }}
+      shell: bash
+
+    - name: Curl Heartbeat
+      run: |
+        echo "Curling LMS heartbeat"
+        status_code=$(curl -s -o /dev/null -w "%{http_code}" http://$LMS_HOST/heartbeat)
+        if [ "$status_code" -ne 200 ]; then
+          echo "Error: LMS Heartbeat endpoint returned status code $status_code"
+          exit 1
+        else
+          echo "Heartbeat endpoint returned status code 200"
+        fi
+      shell: bash
+
+    - name: Set DEMO_COURSE_ID environment variable
+      run: |
+        #TODO: remove this once we stop supporting Tutor <18.0.0
+        if [ "${{ inputs.tutor_version }}" = "<18.0.0" ]; then
+          echo "DEMO_COURSE_ID=course-v1:edX+DemoX+Demo_Course" >> $GITHUB_ENV
+        else
+          echo "DEMO_COURSE_ID=course-v1:OpenedX+DemoX+DemoCourse" >> $GITHUB_ENV
+        fi
+      shell: bash
+
+    - name: Run integration tests
+      if: ${{ inputs.shell_file_to_run }}
+      run: |
+        source .tests_venv/bin/activate
+
+        cd ${{ inputs.app_name }}
+        chmod +x ./${{ inputs.shell_file_to_run }}
+        ./${{ inputs.shell_file_to_run }}
+      shell: bash
+      env:
+        DEMO_COURSE_ID: ${{ env.DEMO_COURSE_ID }}
+        LMS_HOST: ${{ env.LMS_HOST }}
+        CMS_HOST: ${{ env.CMS_HOST }}
```

Comment:
Why do we need to add these env variables again?

Reply:
I was testing something and forgot to remove them. Thanks for noticing!